ngageoint/opensphere
# fix(style): added more options when pushing style events

Now developers can optionally send event(s) on the Source and ColorModel in addition to the Layer.  Also, a starter module for Text has been added; which will reduce the burden when synchronizing and updating recurring text across multiple widgets, buttons, and menu items.

Testing Steps: 

1. Opensphere|http://reflective-grub.surge.sh/opensphere/
2. Add a layer and load some data > Right-click the layer in the Layers modal > Click "Show Features"
3. On the map, Ctrl+Click-and-drag an area inside your loaded data > Click "Select"
4. In the features modal, scroll to where some of the selected features are listed (they will have a green row highlight) > Right-click
5. Verify there are "Color Selected" and "Reset Color" options in the context menu
6. Verify there is an appropriate on-hover tooltip text for both
7. In the Layers modal, Right-click the layer > Click "Feature Actions" > Create an action that changes the color of some of the features (e.g. based on ALT gte 11000)
8. In the Features modal, verify the color dot of the appropriate features has updated
9. On the map, verify the color color dot of the appropriate features has updated
10. In the Feature Actions modal, uncheck the feature action and click "Apply"
11. In the Features modal, verify the color dot of the appropriate features has updated
12. On the map, verify the color color dot of the appropriate features has updated
